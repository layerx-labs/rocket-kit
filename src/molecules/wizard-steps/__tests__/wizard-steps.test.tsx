import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { WizardSteps } from '../../..';
import { Step } from '../';

describe('Wizard Steps', () => {
  it('renders wizard', () => {
    const steps: Step[] = [
      {
        active: true,
        value: 'step 1',
      },
      {
        active: false,
        value: 'step 2',
      },
      {
        active: false,
        value: 'step 3',
      },
    ];

    const { asFragment } = render(<WizardSteps steps={steps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders 3 steps', () => {
    const steps: Step[] = [
      {
        active: true,
        value: 'Step 1',
        dataTestId: 'wizard-1',
      },
      {
        active: false,
        value: 'Step 2',
        dataTestId: 'wizard-2',
      },
      {
        active: false,
        value: 'Step 3',
        dataTestId: 'wizard-3',
      },
    ];

    render(<WizardSteps steps={steps} />);
    for (let step = 1; step < 4; ++step) {
      screen.getByText(`Step ${step}`);
    }
    expect(screen.queryAllByTestId(/step-wizard-/i)).toHaveLength(3);
  });
});
