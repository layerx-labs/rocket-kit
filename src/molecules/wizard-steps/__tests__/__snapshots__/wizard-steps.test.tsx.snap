// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wizard Steps renders wizard 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  width: 100%;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  list-style: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  height: 2.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 1.125rem;
  text-align: center;
}

.c1:before,
.c1:after {
  content: '';
  position: absolute;
  top: 0;
  border: 0 solid #56cad7;
  border-width: 1.375rem 0.9375rem;
  width: 0;
  height: 0;
}

.c1:before {
  left: -0.9375rem;
  border-left-color: transparent;
}

.c1:after {
  left: calc(100% - 1.125rem);
  border-color: transparent;
  border-left-color: #56cad7;
}

.c1:first-child:before {
  border: none;
}

.c1:last-child:after {
  border: none;
}

.c1:first-child {
  overflow: hidden;
}

.c1:first-child > div {
  border-radius: 0.375rem 0 0 0.375rem;
}

.c1:last-child {
  padding-right: 0;
}

.c1:last-child > div {
  border-radius: 0 0.375rem 0.375rem 0;
}

.c1 > div {
  background-color: #56cad7;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > div span {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  font-size: 0.75rem;
  color: #ffffff;
}

.c2 {
  list-style: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  height: 2.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 1.125rem;
  text-align: center;
}

.c2:before,
.c2:after {
  content: '';
  position: absolute;
  top: 0;
  border: 0 solid #f4f4f4;
  border-width: 1.375rem 0.9375rem;
  width: 0;
  height: 0;
}

.c2:before {
  left: -0.9375rem;
  border-left-color: transparent;
}

.c2:after {
  left: calc(100% - 1.125rem);
  border-color: transparent;
  border-left-color: #f4f4f4;
}

.c2:first-child:before {
  border: none;
}

.c2:last-child:after {
  border: none;
}

.c2:first-child {
  overflow: hidden;
}

.c2:first-child > div {
  border-radius: 0.375rem 0 0 0.375rem;
}

.c2:last-child {
  padding-right: 0;
}

.c2:last-child > div {
  border-radius: 0 0.375rem 0.375rem 0;
}

.c2 > div {
  background-color: #f4f4f4;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > div span {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  font-size: 0.75rem;
  color: #939393;
}

@media (min-width:700px) {
  .c1 > div span {
    width: 100%;
    font-size: 1rem;
    font-weight: 700;
  }
}

@media (min-width:700px) {
  .c2 > div span {
    width: 100%;
    font-size: 1rem;
    font-weight: 700;
  }
}

<div
    class="wizard-steps"
    data-testid="wizard-steps-test-id"
  >
    <ul
      class="c0"
    >
      <li
        class="c1"
        data-testid="step-undefined"
      >
        <div>
          <span>
            step 1
          </span>
        </div>
      </li>
      <li
        class="c2"
        data-testid="step-undefined"
      >
        <div>
          <span>
            step 2
          </span>
        </div>
      </li>
      <li
        class="c2"
        data-testid="step-undefined"
      >
        <div>
          <span>
            step 3
          </span>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
